homeassistant/components/matter/entity.py

```diff
@@ -115,10 +115,27 @@ async def async_will_remove_from_hass(self) -> None:
                 # suppress ValueError to prevent race conditions
                 unsub()
 
+    async def async_update(self) -> None:
+        """Call when the entity needs to be updated."""
+        if not self._attr_should_poll:
+            return
+        # manually poll/refresh the primary value
+        await self.matter_client.refresh_attribute(
+            self._endpoint.node.node_id,
+            self.get_matter_attribute_path(self._entity_info.primary_attribute),
+        )
+        self._update_from_device()
+
     @callback
     def _on_matter_event(self, event: EventType, data: Any = None) -> None:
-        """Call on update."""
+        """Call on update from the device."""
         self._attr_available = self._endpoint.node.available
+        if self._attr_should_poll:
+            # secondary attribute updated of a polled primary value
+            # enforce poll of the primary value
+            # debounce it a bit by calling it 2 seconds later
+            self.hass.loop.call_later(2, self.async_schedule_update_ha_state, True)
```

Comment:
to read the energy meter value after the relay was toggled, consider it an extra service to the user otherwise it looks weird when you toggle the relay you wait up to 60 second to see the wattage change

Reply:
What Martin is saying still holds: how to avoid scheduling 5 polls when 5 events come in within a second.